### DISCUSSION TITLE
How to render a 2D projected map with SkyNode

### DISCUSSION THREAD

**Discussion by chenguanga** [pelicanmapping/osgearth#1895] [Support Q&A] [answered]
**I want to create a 2D projected map,but the SkyNode cannot be rendered correctly, main code is as below:**

        MapNode* mapNode  = (MapNode* )osgDB::readNodeFile("openstreetmap_flat.earth");
        osgEarth::Util::SkyNode* skyNode = osgEarth::Util::SkyNode::create();
        skyNode->addChild(mapNode);
        osgViewer::Viewer viewer;
        skyNode->attach(&viewer);
        viewer.setSceneData( skyNode);    

**Although 2D map had been created, the atmosphere of SkyNode was not rendered correctly. I try to find answer in the source code:**

    if (mapNode->getMapSRS()->isProjected())
    {
        osgEarth:: GeoPoint refPoint =
            mapNode->getMap()->getProfile()->getExtent().getCentroid();
        skyNode->setReferencePoint(refPoint);
    }
**The SkyNode is still keeped its 3D sphere shape in the scene, are there any relevant examples or code blocks can be used for reference?**
![image](https://user-images.githubusercontent.com/36613682/145022897-e8d02216-8546-47a0-a7e5-ef3fbcb0ec4e.png)

**Comment by gwaldron** [comment, marked as answer]
`SimpleSkyNode` only supports geodetic maps. You can try using the `GLSkyNode` for a projected map.
